packages/core/src/html/codeblockProcessor.js

```diff
@@ -129,6 +129,28 @@ function highlightCodeBlock(node) {
   });
 }
 
+/**
+ * Adjust the class attribute of code blocks according to the global line numbers setting.
+ * This looks into the class of each code block and if no `line-numbers` or `no-line-numbers` found,
+ * append the respective class.
+ * @param node the code block element, which is the 'code' node
+ * @param showCodeLineNumbers true if line numbers should be shown, false otherwise
+ */
+function setCodeLineNumbers(node, showCodeLineNumbers) {
+  const existingClass = node.attribs.class;
+  if (existingClass
+    && (existingClass.includes('line-numbers') || existingClass.includes('no-line-numbers'))) {
```

Comment:
The first check here also tests for the second since it is a string, unlike `classList` in the browser.

Perhaps we could make this a little more robust with a regex in that sense too. (avoid something like `foo-line-numbers`)